packages/lesswrong/components/ea-forum/voting-portal/EAVotingPortalSelectCandidatesPage.tsx

```diff
@@ -30,25 +28,41 @@ const EAVotingPortalSelectCandidatesPageLoader = ({ classes }: { classes: Classe
 };
 
 const EAVotingPortalSelectCandidatesPage = ({
-  selectedCandidateIds,
   electionVote,
   updateVote,
   classes,
 }: {
-  selectedCandidateIds: string[];
   electionVote: ElectionVoteInfo;
   updateVote: (newVote: NullablePartial<DbElectionVote>) => Promise<void>;
   classes: ClassesType;
 }) => {
   const { ElectionCandidatesList, VotingPortalFooter } = Components;
-  const [selectedIds, setSelectedCandidateIds] = useState<string[]>(selectedCandidateIds);
+  const [selectedIds, setSelectedCandidateIds] = useState<string[]>(Object.keys(electionVote.vote));
   const [totalCount, setTotalCount] = useState<number>(0);
   const navigate = useNavigate();
+  const { flash } = useMessages();
 
   const saveSelection = useCallback(async () => {
-    const newVote = selectedIds.reduce((acc, id) => ({ ...acc, [id]: electionVote.vote[id] ?? null }), {})
-    await updateVote({vote: newVote});
-  }, [electionVote, selectedIds, updateVote]);
+    if (selectedIds.length === 0) {
+      flash("You must select at least one candidate");
+      return;
+    }
+    const isSingleCandidate = selectedIds.length === 1;
+
+    try {
+      const newVote = isSingleCandidate
+        ? { [selectedIds[0]]: 1 } // If there is only one candidate, give them all the votes
+        : selectedIds.reduce((acc, id) => ({ ...acc, [id]: electionVote.vote[id] ?? null }), {});
+      // In case they are redoing this step, clear the compare state
+      await updateVote({vote: newVote, compareState: null});
```

Comment:
I feel like ideally this would not clear the compare state. Like if I just removed one of the candidates, I'd still want to see my other comparison values in the next step. Prob not worth fixing though, prob an edge case.